src/freenas/etc/directoryservice/LDAP/ctl

```diff
@@ -151,7 +151,7 @@ ldapctl_start()
 	fi
 
 	anonbind="$(LDAP_get ldap_anonbind)"
-	if [ "${anonbind}" = "0" ]
+	if [ "${anonbind}" = "0" -o -n "${realm}" -o -n "${keytab_principal}" ]
```

Comment:
You are saying "if not an anonymous bind, OR realm is not null OR keytab principal is not null" here. Is this really what you want? I don't think you are accomplishing what you are trying to do here. anonbind will be 1 when anonymous binding is used for one, also, you will want to check 2 cases for anonymous binding here. You will want to check anonymous binding with kerberos, and without kerberos.

Reply:
Well, the goal is basically two configuration options:
1) sssd (we use this if we're not using anonymous bind or if we're using kerberos for auth)
2) nslcd (we use this if we're only doing an anonymous bind)

Big picture is to configure sssd in this specific situation to use LDAP for an ID provider, but kerberos for auth.